crates/matrix-sdk-base/src/client.rs

```diff
@@ -384,6 +384,7 @@ impl BaseClient {
         room: &Room,
         limited: bool,
         events: Vec<Raw<AnySyncTimelineEvent>>,
+        ignore_state_events: bool,
```

Comment:
Not a fan of this boolean, but ok it's not as complex as I imagined it would be.

Reply:
Me neither! I'm gonna refactor this in a next PR (see the PR description where I already mention this).